pbj-core/pbj-compiler/src/main/java/com/hedera/pbj/compiler/impl/generators/ModelGenerator.java

```diff
@@ -363,6 +361,58 @@ public int hashCode() {
 		return bodyContent;
 	}
 
+	/**
+	 * Generates a pre-populated constructor for a class.
+	 * @param fields the fields to use for the code generation
+	 * @return the generated code
+	 */
+	private static String generateConstructor(
+			final String constructorName,
+			final List<Field> fields,
+			final boolean shouldThrowOnOneOfNull,
+			final MessageDefContext msgDef,
+			final ContextualLookupHelper lookupHelper) {
+		if (fields.isEmpty()) {
+			return "";
+		}
+		return """
+			    /**
+			     * Create a pre-populated $constructorName.
+			     * $constructorParamDocs
+			     */
+			    public $constructorName($constructorParams) {
+			$constructorCode    }
+			"""
+				.replace("$constructorParamDocs",fields.stream().map(field ->
+						"\n     * @param "+field.nameCamelFirstLower()+" "+
+								field.comment().replaceAll("\n", "\n     *         "+" ".repeat(field.nameCamelFirstLower().length()))
+				).collect(Collectors.joining(", ")))
+				.replace("$constructorName", constructorName)
+				.replace("$constructorParams",fields.stream().map(field ->
+						field.javaFieldType() + " " + field.nameCamelFirstLower()
+				).collect(Collectors.joining(", ")))
+				.replace("$constructorCode",fields.stream().map(field -> {
+					StringBuilder sb = new StringBuilder();
+					if (shouldThrowOnOneOfNull && field instanceof OneOfField) {
+						sb.append(generateConstructorCodeForField(field)).append('\n');
+					}
+					switch (field.type()) {
+						case BYTES, STRING: {
+							sb.append("this.$name = $name != null ? $name : $default;"
+									.replace("$name", field.nameCamelFirstLower())
+									.replace("$default", getDefaultValue(field, msgDef, lookupHelper))
+							);
+							break;
+						}
+						default:
+							sb.append("this.$name = $name;".replace("$name", field.nameCamelFirstLower()));
+							break;
+					}
+					return sb.toString();
+				}).collect(Collectors.joining("\n")).indent(DEFAULT_INDENT * 2))
+				;//.indent(DEFAULT_INDENT);
```

Comment:
please, clean it up.

Reply:
Good catch. I'll do this a follow-up PR when I'm re-applying the `maxDepth` fix later today.